lib/3scale/backend/job_fetcher.rb

```diff
@@ -85,6 +85,8 @@ def try_pop_from_queue(max=nil)
       def wait_pop_from_queue
         queue, encoded = @redis.blpop(*@queues, timeout: @fetch_timeout)
         [queue, [encoded]] if encoded
+      rescue RedisClient::ReadTimeoutError => _e
+        # Do nothing, we expect frequent timeouts from the sync client when the queue is empty
```

Comment:
Could you then change the comment to clarify that this is a bug with sentinels and we should log these errors once it is fixed? Or in case it is easy to know that this sentinel or not, log for the cases where sentinels are not used?

Reply:
> Could you then change the comment to clarify that this is a bug with sentinels and we should log these errors once it is fixed?

OK

> Or in case it is easy to know that this sentinel or not, log for the cases where sentinels are not used?

Detecting sentinels from here is possible but too messy, I'd prefer to not do it. About logging, do you mean for `RedisClient::ReadTimeoutError` or for any error?

Reply:
About logging I mean `RedisClient::ReadTimeoutError`. Would be good to log any connectivity errors, but it will probably require more investigation. And I expect anything else to be pretty rare anyway so raising full blown should be ok.